library/core/api/src/main/kotlin/de/lise/fluxflow/api/step/stateful/data/StepDataService.kt

```diff
@@ -6,4 +6,13 @@ interface StepDataService {
     fun getData(step: Step): List<Data<*>>
     fun <T> getData(step: Step, kind: DataKind): Data<T>?
     fun <T> setValue(data: Data<T>, value: T)
+
+    /**
+     * Assigns the given [value] to the [data] while also providing extra [context] information.
+     *
+     * @param context Arbitrary information to be passed along.
+     * @param value The value to be assigned.
+     * @param data The data to be updated.
+     */
+    fun <T> setValue(context: Any, data: Data<T>, value: T)
```

Comment:
Is there a specific reason why context is the first parameter? Wouldn't it be more consistent if the order of params stays the same in this overload?

Reply:
+1

Reply:
We chose this order on purpose. For further information contact @masinger